daffodil-lib/src/main/scala/org/apache/daffodil/exceptions/Assert.scala

```diff
@@ -33,7 +33,7 @@ abstract class ThinException protected (dummy: Int, cause: Throwable, fmt: Strin
   extends Exception(null, cause, false, false) {
 
   private lazy val msg_ =
-    if (fmt ne null) fmt.format(args)
+    if (fmt ne null) fmt.format(args: _*)
```

Comment:
I can see how this particular piece of list/argsList polymorphism issometimes useful, but generally, I wish this could be a scala type error so that it's caught at compile time, not run time. Oh well.

Reply:
Yeah, I think it's just a genearl issue with varargs. Those functions can accept anything, and it's hard to distinguish between a a single list as being the argument or the individual fields being all the arguments.